src/Compilers/CSharp/Portable/Binder/Binder_Symbols.cs

```diff
@@ -1986,12 +1986,19 @@ public int Compare(Symbol fst, Symbol snd)
                                 }
                                 else if (srcSymbol.Kind == SymbolKind.NamedType && mdSymbol.Kind == SymbolKind.NamedType)
                                 {
-                                    // WRN_SameFullNameThisAggAgg: The type '{1}' in '{0}' conflicts with the imported type '{3}' in '{2}'. Using the type defined in '{0}'.
-                                    diagnostics.Add(ErrorCode.WRN_SameFullNameThisAggAgg, where.Location, originalSymbols,
-                                        arg0,
-                                        srcSymbol,
-                                        mdSymbol.ContainingAssembly,
-                                        mdSymbol);
+                                    // It is common for source generators to generate internal attributes to drive generator behavior. These are then visible to
+                                    // IVT assemblies that may also run the same source generator. We don't want to warn in these cases, so we suppress the warning
+                                    // if the source symbol is from a source-generated file. We only do this for types with a single location, since that is the very
+                                    // specific scenario we are targeting.
+                                    if (srcSymbol.Locations is not [{ SourceTree: { } srcTree }] || !Compilation.IsSourceGeneratedTree(srcTree))
```

Comment:
It feels wrong to me to tie this to generated code because it doesn't satisfy all the scenarios. Consider: 

1. A key scenario for source generators is that you can _walk away from them_. Essentially at any point you can set `<EmitGeneratedCompilerFiles>true`, check in the files, remove the generators and move on with life. Tieing the diagnostic to this heuristic means that scenario breaks 
2. This problem comes up frequently in non-generated code. Consider that there are many NuPkg out there which distribute source not DLLs. They have this exact problem and it's not generated code, it's plain old code. 

My instinct is to lean towards a solution where a type is marked as 

> I know there may be multiple definitions of me, so what, pick the source.